oneview/resource_logical_interconnect_group.go

```diff
@@ -1720,110 +1708,78 @@ func resourceLogicalInterconnectGroupRead(d *schema.ResourceData, meta interface
 
 	d.Set("interconnect_map_entry_template", interconnectMapEntryTemplates)
 
-	uplinkSets := make([]map[string]interface{}, 0, len(logicalInterconnectGroup.UplinkSets))
-	for i, uplinkSet := range logicalInterconnectGroup.UplinkSets {
-
-		primaryPortEnclosure := 0
-		primaryPortBay := 0
-		primaryPortPort := 0
+	//Reading UplinkSets
+	emptyUplinkSets := []ov.UplinkSets{}
+	if !reflect.DeepEqual(logicalInterconnectGroup.UplinkSets, emptyUplinkSets) {
+		uplinkSets := make([]map[string]interface{}, 0)
+		if len(logicalInterconnectGroup.UplinkSets) != 0 {
+			for i, uplinkSet := range logicalInterconnectGroup.UplinkSets {
+
+				uplinkSets = append(uplinkSets, map[string]interface{}{
+					"network_type":          uplinkSet.NetworkType,
+					"ethernet_network_type": uplinkSet.EthernetNetworkType,
+					"name":                  uplinkSet.Name,
+					"mode":                  uplinkSet.Mode,
+					"lacp_timer":            uplinkSet.LacpTimer,
+				})
 
-		if uplinkSet.PrimaryPort != nil {
-			for _, primaryPortLocation := range uplinkSet.PrimaryPort.LocationEntries {
-				if primaryPortLocation.Type == "Bay" {
-					primaryPortBay = primaryPortLocation.RelativeValue
-				}
-				if primaryPortLocation.Type == "Enclosure" {
-					primaryPortEnclosure = primaryPortLocation.RelativeValue
-				}
-				if primaryPortLocation.Type == "Port" {
-					primaryPortPort = primaryPortLocation.RelativeValue
+				if uplinkSet.NativeNetworkUri.String() != "null" {
+					uplinkSets[i]["native_network_uri"] = uplinkSet.NativeNetworkUri.String()
 				}
-			}
-		}
 
-		logicalPortConfigs := make([]map[string]interface{}, 0, len(uplinkSet.LogicalPortConfigInfos))
-		for _, logicalPortConfigInfo := range uplinkSet.LogicalPortConfigInfos {
-			portEnclosure := 0
-			portBay := 0
-			portPort := 0
-			primaryPort := false
-			for _, portLocation := range logicalPortConfigInfo.LogicalLocation.LocationEntries {
-				if portLocation.Type == "Bay" {
-					portBay = portLocation.RelativeValue
-				}
-				if portLocation.Type == "Enclosure" {
-					portEnclosure = portLocation.RelativeValue
-				}
-				if portLocation.Type == "Port" {
-					portPort = portLocation.RelativeValue
+				// Collecting primary location details
+				primaryPortEnclosure := 0
```

Comment:
Not quite sure about these fields, do we really need those now?

Reply:
as part of configuration I've added it because it was their earlier to provide backward compatibility. Not sure if few customer would be using it or not.